app/routes/_gcn.circulars.$circularId.($version)/route.tsx

```diff
@@ -0,0 +1,185 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { HeadersFunction, LoaderFunctionArgs } from '@remix-run/node'
+import { json } from '@remix-run/node'
+import { Link, useLoaderData, useRouteLoaderData } from '@remix-run/react'
+import { Button, ButtonGroup, CardBody, Icon } from '@trussworks/react-uswds'
+import { useState } from 'react'
+import invariant from 'tiny-invariant'
+
+import type { loader as parentLoader } from '../_gcn.circulars.$circularId/route'
+import { get } from '../_gcn.circulars/circulars.server'
+import { PlainTextBody } from './Body'
+import { FrontMatter } from './FrontMatter'
+import DetailsDropdownButton from '~/components/DetailsDropdownButton'
+import DetailsDropdownContent from '~/components/DetailsDropdownContent'
+import { feature, origin } from '~/lib/env.server'
+import {
+  getCanonicalUrlHeaders,
+  pickHeaders,
+  publicStaticShortTermCacheControlHeaders,
+} from '~/lib/headers.server'
+import { useSearchString } from '~/lib/utils'
+import { useFeature } from '~/root'
+import type { BreadcrumbHandle } from '~/root/Title'
+
+export const handle: BreadcrumbHandle<typeof loader> = {
+  breadcrumb({ data }) {
+    if (data) {
+      const { subject, version } = data
+      return `${version ? `v${version} - ` : ''} ${subject}`
+    }
+  },
+}
+
+export async function loader({
+  params: { circularId, version },
+}: LoaderFunctionArgs) {
+  if (!feature('CIRCULAR_VERSIONS') && version)
+    throw new Response(null, { status: 404 })
+  invariant(circularId)
+  const result = await get(
+    parseFloat(circularId),
+    version ? parseFloat(version) : undefined
+  )
+
+  return json(result, {
+    headers: {
+      ...publicStaticShortTermCacheControlHeaders,
+      ...getCanonicalUrlHeaders(new URL(`/circulars/${circularId}`, origin)),
+    },
+  })
+}
+
+export const headers: HeadersFunction = ({ loaderHeaders }) =>
+  pickHeaders(loaderHeaders, ['Link'])
+
+export default function () {
+  const { circularId, body, bibcode, version, ...frontMatter } =
+    useLoaderData<typeof loader>()
+  const searchString = useSearchString()
+
+  const result = useRouteLoaderData<typeof parentLoader>(
```

Comment:
```suggestion
  const { history, userIsModerator } = useRouteLoaderData<typeof parentLoader>(
```

Reply:
Yes, but once you remove the canonical URL thing in the parent route, it should work.

Reply:
I see, I had removed the headers, but the json function was still there. Removing now

Reply:
I actually still get this error without the json part:
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/fbe1ccc2-1cb8-497f-9c98-065ca163d606)

Reply:
OK, I'll take a look.

Reply:
From our other uses of the function it looks like we use it to define a const, then access it's properties

Reply:
Oh, I see... it's because `useRouteLoaderData` is allowed to return `undefined`.


```ts
/**
 * Returns the JSON parsed data from the current route's `loader`.
 *
 * @see https://remix.run/hooks/use-loader-data
 */
export declare function useLoaderData<T = AppData>(): SerializeFrom<T>;
/**
 * Returns the loaderData for the given routeId.
 *
 * @see https://remix.run/hooks/use-route-loader-data
 */
export declare function useRouteLoaderData<T = AppData>(routeId: string): SerializeFrom<T> | undefined;
```